api_generator/lib/api_generator.rb

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+# frozen_string_literal: true
+
+require 'openapi3_parser'
+require_relative 'action'
+require_relative 'action_generator'
+require_relative 'spec_generator'
+require_relative 'namespace_generator'
+require_relative 'index_generator'
+
+# Generate API endpoints for OpenSearch Ruby client
+class ApiGenerator
+  HTTP_VERBS = %w[get post put patch delete patch head].freeze
+  EXISTING_NAMESPACES = Set.new(%w[
```

Comment:
Externalize into a set of folders/files (e.g. load a list of folders checked into git).